tests/Config/ConfigTest.php

```diff
@@ -22,15 +22,12 @@
 
 class ConfigTest extends \Psalm\Tests\TestCase
 {
-    /** @var TestConfig */
-    protected static $config;
-
     /** @var \Psalm\Internal\Analyzer\ProjectAnalyzer */
     protected $project_analyzer;
 
     public static function setUpBeforeClass() : void
     {
-        self::$config = new TestConfig();
+        new TestConfig();
```

Comment:
Nope, since everything inherits from `TestCase` and there's now an instance property `protected $config;` (non-static), these changes were necessary.
`new TestConfig()` in the parts you pointed out was used to initialize their internal static state - `self:$config` basically had not meaning at all and was never used/retrieved.

Reply:
However, I could have created a separate clean-up commit to make that more explicit.

Reply:
I'll revert those changes and introduce a new `protected TestCase::$testConfig` variable instead...